### ISSUE TITLE
Not working with electron 2.0

### ISSUE THREAD

**Issue by Stanzilla** [electron/electron-rebuild#240]
```
An unhandled error occurred inside electron-rebuild
Could not detect abi for version 2.0.0 and runtime electron.  Updating "node-abi" might help solve this issue if it is a new release of electron

Error: Could not detect abi for version 2.0.0 and runtime electron.  Updating "node-abi" might help solve this issue if it is a new release of electron
    at Object.getAbi (\node_modules\node-abi\index.js:30:9)
    at new Rebuilder (\node_modules\electron-rebuild\lib\src\rebuild.js:64:28)
    at rebuildWithOptions (\node_modules\electron-rebuild\lib\src\rebuild.js:310:23)
    at Object.doRebuild [as rebuild] (\node_modules\electron-rebuild\lib\src\rebuild.js:317:16)
    at Object.<anonymous> (\node_modules\electron-rebuild\lib\src\cli.js:114:33)
    at Generator.next (<anonymous>)
    at \node_modules\electron-rebuild\lib\src\cli.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (\node_modules\electron-rebuild\lib\src\cli.js:4:12)
    at \node_modules\electron-rebuild\lib\src\cli.js:69:8
```

Update to node-abi 2.4.0 should fix it, https://github.com/lgeiger/node-abi/releases/tag/v2.4.0

**Comment by malept** [comment]
Thanks for filing an issue, but according to the `package.json`, it already works with `node-abi` 2.4.0:

https://github.com/electron/electron-rebuild/blob/90fe7fc21cffca5abff58cbedf455dcf0aaf815e/package.json#L38

You should just be able to update the dependencies of `electron-rebuild`  (or reinstall it) and you should get the updated dependency.

**Comment by fangzhzh** [comment]
I'm on latest version, 1.8.1 and still get the same problem
```
An unhandled error occurred inside electron-rebuild
Could not detect abi for version 2.0.0 and runtime electron.  Updating "node-abi" might help solve this issue if it is a new release of electron

Error: Could not detect abi for version 2.0.0 and runtime electron.  Updating "node-abi" might help solve this issue if it is a new release of electron
    at Object.getAbi (/Users/developer/app/node_modules/node-abi/index.js:44:9)
    at new Rebuilder (/Users/developer/app/node_modules/electron-rebuild/lib/src/rebuild.js:64:28)
    at rebuildWithOptions (/Users/developer/app/node_modules/electron-rebuild/lib/src/rebuild.js:315:23)
    at Object.doRebuild [as rebuild] (/Users/developer/app/node_modules/electron-rebuild/lib/src/rebuild.js:322:16)
    at Object.<anonymous> (/Users/developer/app/node_modules/electron-rebuild/lib/src/cli.js:114:33)
    at Generator.next (<anonymous>)
    at /Users/developer/app/node_modules/electron-rebuild/lib/src/cli.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/developer/app/node_modules/electron-rebuild/lib/src/cli.js:4:12)
    at /Users/developer/app/node_modules/electron-rebuild/lib/src/cli.js:69:8
```

**Comment by Siilwyn** [comment]
Hey @Stanzilla did you get this fixed by updating `electron-rebuild`?

Updated `electron` to `2.0.4` and `electron-rebuild` to `1.8.1` resulting in the error: `Could not detect abi for version 2.0.4 and runtime electron.`.

**Comment by malept** [comment]
@Siilwyn according to your `package-lock.json`, `node-abi` is at 2.2.0.

**Comment by Siilwyn** [comment]
@malept sorry to bother you again but what does that mean? That I need to downgrade it or do I need 2.4.0?

**Comment by malept** [comment]
You need [`node-abi` 2.4.1](https://github.com/lgeiger/node-abi/releases/tag/v2.4.1) or later.

**Comment by Siilwyn** [comment]
Ah, thank you for your reply!

This is unexpected npm behaviour, I expected `npm install electron-rebuild@latest` to also get the latest nested dependencies, had to remove the lock **and** `node_modules`.

**Comment by christianscheuer** [comment]
In using `lerna` and `yarn` as package managers, `electron-rebuild`s dependency on `node-abi@^2.0.0` resolves to `node-abi@2.2.0`, but as stated above, node-abi@2.4.1 is required to build for electron 2.0+. 

Even when using a direct dependency on node-abi@2.4.3, electron-rebuild is still getting the 2.2.0 version.

Would it be possible for electron-rebuild to update its dependency to 2.4.3 (which is the combination that actually works) so we won't have to make ugly workarounds here? Or is there something else I would need to configure that is not apparent?
This has worked just fine with the old releases, and works fine when using a "resolutions" workaround in yarn, but since lerna doesn't support that, we have to hack our build phase to make it work, which is not pretty.

**Comment by malept** [comment]
It seems to me that if a package manager can't let you update a dependency (even a dependency of a dependency) when its semver definition allows it, that's a fairly large feature missing in that package manager. For example, NPM and yarn allow you to `npm update node-abi` and `yarn upgrade node-abi`, respectively.

**Comment by christianscheuer** [comment]
@malept thanks. You're completely right, of course. My apologies. When creating a fresh repo with just that single dependency on electron-rebuild@1.8.1, it resolves just fine to the node-abi@2.4.3, so something must have been off in either some local state, or maybe a yarn bug combining dependencies. In any case not something relevant for this repo, I agree.
